core/src/nucleus/ribosome/api/init_globals.rs

```diff
@@ -25,7 +27,16 @@ pub fn invoke_init_globals(runtime: &mut Runtime, _args: &RuntimeArgs) -> ZomeAp
         agent_initial_hash: HashString::from(""),
         agent_latest_hash: HashString::from(""),
         public_token: Address::from(""),
-        cap_request: runtime.zome_call_data()?.call.cap.clone(),
+        cap_request: runtime
+            .zome_call_data()
+            .map(|zome_call_data| zome_call_data.call.cap.clone())
+            .unwrap_or_else(|_| {
+                CapabilityRequest::new(
+                    Address::from("none"),
+                    Address::from("none"),
+                    Signature::fake(),
+                )
```

Comment:
@zippy: I'm just creating a bogus cap_request here. Should that field be an `Option<CapabilityRequest>` so we can set it to `None` in case we are in a callback...

Reply:
The context here is that the the app should be able to find the provenance of who made a given call, and that's inside the capabilityrequest of the call.  So yes, I think we should use None to indicate the context that the call came from self as a callback.  I guess we could have some enum to make this more explicit.  But we definitely shouldn't have a bogus CapabilityRequest.